subsys/bluetooth/controller/ll_sw/ull_iso.c

```diff
@@ -58,8 +61,30 @@ static int init_reset(void);
 #if BT_CTLR_ISO_STREAMS
 static struct ll_iso_datapath datapath_pool[BT_CTLR_ISO_STREAMS];
 #endif
+
+static void iso_rx_demux(void *param);
 static void *datapath_free;
 
+#if defined(CONFIG_BT_CTLR_SYNC_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+#define NODE_RX_HEADER_SIZE (offsetof(struct node_rx_pdu, pdu))
+/* ISO LL conformance tests require a PDU size of maximum 251 bytes */
+#define ISO_RX_BUFFER_SIZE 251
+
+/* Declare the ISO rx node RXFIFO. This is a composite pool-backed MFIFO for
+ * rx_nodes. The declaration constructs the following data structures:
+ * - mfifo_iso_rx:    FIFO with pointers to PDU buffers
+ * - mem_iso_rx:      Backing data pool for PDU buffer elements
+ * - mem_link_iso_rx: Pool of memq_link_t elements
+ *
+ * Two extra links are reserved for use by the ll_iso_rx and ull_iso_rx memq.
+ */
+static RXFIFO_DEFINE(iso_rx, NODE_RX_HEADER_SIZE + ISO_RX_BUFFER_SIZE,
+			     CONFIG_BT_CTLR_ISO_RX_BUFFERS, 2);
+
+static MEMQ_DECLARE(ll_iso_rx);
+static MEMQ_DECLARE(ull_iso_rx);
```

Comment:
This `ull_iso_rx` is only required for proprietary path, right? Else, Rx-ed ISO payload can be directly handed over to ISOAL in thread context, right?

Reply:
Yes, it could be done like that. However, in our implementation RX'ed ISO payload is received in LLL context and travels up via this MEMQ for both HCI- and proprietary data path. How would you otherwise get payloads from RX context to ULL and HCI, unless you use the normal RX nodes?

Reply:
OK I think I get it now. Is it so that you want to use `ll_iso_rx_put` in interrupt context, and thereby skip the ULL queue, `iso_rx_demux` etc.? But then you would have to deliver data via different calls, if you want to switch data path runtime.

Reply:
yes, to directly enqueue to thread using `ll_iso_rx_put`.

Only when proprietary data path is supported, we should be able to switch the data path in ULL context, but if only HCI data path is supported, then let `ll_iso_rx_put` be called from LLL.

I want to avoid redundant ULL CPU use when only HCI data path is supported in the build. Similarly, if only proprietary data path is used, the rx buffers could be owned by ULL (reducing thread context CPU use for releasing rx buffers) ?